### DISCUSSION TITLE
Support `py.ini`

### DISCUSSION THREAD

**Discussion by ssbarnea** [brettcannon/python-launcher#271] [Ideas]
While official Python documentation mentions where `py.ini` can be loaded from and what you can configure inside, the unix python-launcher does not see to have any support for it, or lacks documentation.

I reached this while trying to configure `py` on macos to prefer using the latest release of python instead of using the latest pre-release (which obviously does not work with most tools).

```
py --list
 3.12 │ /usr/local/bin/python3.12    
 3.11 │ /opt/homebrew/bin/python3.11 
 3.10 │ /opt/homebrew/bin/python3.10 
 3.9  │ /opt/homebrew/bin/python3.9  
 3.8  │ /opt/homebrew/bin/python3.8  
```

I created a `py.ini` in user home with expected config but it was not loaded. Should I assume that python-launcher already knowns to use XDG standard. (XDG_CONFIG_HOME) and look for confing in `~/.config/py.ini` or similar? I tried to run with verbose mode but apparently `--list` does not even accept that argument.
```
$ py --list -vv      
The `--list` flag must be specified on its own; see `py --help` for details
FAIL: 64
```

Later I found that `export PYLAUNCH_DEBUG=1` added some extra debugging info and I attempted to trick py to default to another version of python using the documented environment variables:
```
export PY_PYTHON=3.11
export PY_PYTHON3=3.11
```
While this works, I still think that lack of config file support is a bug, probably the missing `-v` too.

**Comment by brettcannon** [comment]
> While official Python documentation mentions where `py.ini` can be loaded from and what you can configure inside, the unix python-launcher does not see to have any support for it, or lacks documentation.
...
> I still think that lack of config file support is a bug, probably the missing `-v` too.

[`py.ini` files](https://docs.python.org/3/using/windows.html#customization-via-ini-files) are not supported on purpose (hence why it's not documented at https://python-launcher.app/ or in the man page), so it's not a bug. I personally dislike the INI file format and honestly no one has ever asked for this before. Plus it seemed slightly redundant compared to the environment variables, so I didn't worry about it (Windows has a somewhat special need for this because environment variables are way harder to set there than on Unix).

I'm hoping something like https://github.com/brettcannon/python-launcher/discussions/165 will replace the need of `py.ini`. But if people want a similar mechanism as `py.ini` then I encourage them to upvote this feature request.

As for `-v`, I don't see [that supported](https://github.com/python/cpython/blob/b724ac2fe7fbb5a7a33d639cad8e748f17b325e0/PC/launcher2.c#L611-L654) by the Windows Launcher.

**Reply by brettcannon** [reply]
I have also updated the docs at https://python-launcher.app/ to have an admonition explaining the differences with the Python Launcher for Windows.
